synapse/crypto/keyring.py

```diff
@@ -657,9 +656,10 @@ def get_server_verify_key_v2_indirect(self, keys_to_fetch, key_server):
                 },
             )
         except (NotRetryingDestination, RequestSendFailed) as e:
-            raise_from(KeyLookupError("Failed to connect to remote server"), e)
+            # these both have str() representations which we can't really improve upon
+            raise KeyLookupError(str(e))
         except HttpResponseException as e:
-            raise_from(KeyLookupError("Remote server returned an error"), e)
```

Comment:
Its worth noting that in py3 adding an exception as context happens implicitly when raising from within an except block, aiui.

Reply:
oh rly?